src/Microsoft.SqlTools.ServiceLayer/LanguageServices/ConnectedBindingContext.cs

```diff
@@ -192,8 +200,13 @@ private static DatabaseCompatibilityLevel GetDatabaseCompatibilityLevel(ServerVe
         /// Gets the transaction sql version from a server version
         /// </summary>
         /// <param name="serverVersion"></param>
-        private static TransactSqlVersion GetTransactSqlVersion(ServerVersion serverVersion)
+        private static TransactSqlVersion GetTransactSqlVersion(DatabaseEngineType engineType, ServerVersion serverVersion)
```

Comment:
We could make this somewhat more resilient to SQL editions going rogue (eg MI) by getting the actual compatibility level of the database and using max(version, actualcompatibilitylevel) for non-azure instances.

Reply:
Alright, took a shot at that. @shueybubbles what do you think?